presto-main/src/main/java/io/prestosql/cost/PlanCostEstimate.java

```diff
@@ -26,12 +26,13 @@
 
 public final class PlanCostEstimate
 {
-    private static final PlanCostEstimate INFINITE = new PlanCostEstimate(POSITIVE_INFINITY, POSITIVE_INFINITY, POSITIVE_INFINITY);
-    private static final PlanCostEstimate UNKNOWN = new PlanCostEstimate(NaN, NaN, NaN);
-    private static final PlanCostEstimate ZERO = new PlanCostEstimate(0, 0, 0);
+    private static final PlanCostEstimate INFINITE = new PlanCostEstimate(POSITIVE_INFINITY, POSITIVE_INFINITY, POSITIVE_INFINITY, POSITIVE_INFINITY);
+    private static final PlanCostEstimate UNKNOWN = new PlanCostEstimate(NaN, NaN, NaN, NaN);
+    private static final PlanCostEstimate ZERO = new PlanCostEstimate(0, 0, 0, 0);
 
     private final double cpuCost;
-    private final double memoryCost;
+    private final double maxMemory;
```

Comment:
this should be called `peakMemory` and `peakMemoryWhenOutputting`

Reply:
i initially had peakMemory but though maxMemory describes estimation's intent better.